### PR TITLE
Historian formula

### PR DESCRIPTION
**Pull request by usrbinpikachu**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/historian.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class Historian < Formula
+  desc "Command-line utility for managing shell history in a SQLite database."
+  homepage "https://github.com/jcsalterego/historian"
+  url "https://files.jakemcknight.com/historian-0.1.tar.gz"
```
Comment (bfontaine) [inline Formula/historian.rb]: We need an official/tagged release from upstream here.

**Review by bfontaine — CHANGES_REQUESTED** [review]

File: Formula/historian.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  sha256 "80981cf45b6c51dd2969a6d5cd2d69a488e9a58a411fc51d5bee68aaf267d7a5"
+
+  depends_on "sqlite"
+
+  def install
+    bin.install "hist"
+  end
+
+  test do
+    system "/usr/local/bin/hist", "version"
```
Comment (bfontaine) [inline Formula/historian.rb]: Don’t hardcode the path; some people don’t have Homebrew under `/usr/local`. Use `system bin/"hist"`. Also, can’t we write a test that does something more substantial than that? You could create a dummy `.bash_history` then use `hist` to import it.

**Comment by kudakeru** [comment]
@bfontaine Thanks for the suggestions. The formula now uses the tagged release from its repo, and has a much better test.

**Comment by bfontaine** [comment]
Thanks! However you can’t add a test file to Homebrew like so. You have to write it in the test block, like so:
```ruby
test do
  (testpath/".bash_history").write <<-EOS.undent
    your file
    content
    goes here
  EOS

  # ... your test ...
end
```

**Review by bfontaine — APPROVED** [review]

File: .bash_history [hunk 3]
```diff
@@ -0,0 +1,6 @@
+ssh-keygen -t rsa -b 4096
+cat id_rsa.pub
+pbcopy < id_rsa.pub
+cd ~
+vim ~/.ssh/config
+brew update
```
Comment (bfontaine) [inline .bash_history]: Please remove this whole `.bash_history` file; it’s not needed.

Comment (kudakeru) [inline .bash_history]: Ah, my mistake. Fixed now.

**Comment by bfontaine** [comment]
@BrewTestBot test this please.

**Comment by tdsmith** [comment]
Thanks @kudakeru for your contribution, and thanks @bfontaine for shepherding it through!

**Pull request closed without merging**